### ISSUE TITLE
Rotate pivot group

### ISSUE THREAD

**Issue by microcipcip** [WonderMediaProductions/Maya2glTF#40]
First, thanks for your awesome plugin, it is really easy to use! 

I have a problem regarding the mesh pivot point and not sure if it is an issue with this plugin or Three.JS (or probably with me). Maybe you can point me to the right direction and the right Maya setup I should use? 

I am basically trying to rotate the hand of a model (with the fingers nested inside it) and keep it somehow attached to the arm, but I can't manage to. [Here is the full question on threejs forum](https://discourse.threejs.org/t/rotate-group-around-pivot/3656).

**Comment by Ziriax** [comment]
For some reason I did not get a notification of this issue, sorry for the late reply.

I do have some test scenes for pivot points, let me first double check these still export correctly.

**Comment by Ziriax** [comment]
Indeed, it seems the pivot doesn't export correctly anymore.

I really should find some time to add regression tests :-)

I will look into this immediately.

Thanks for reporting!

**Comment by microcipcip** [comment]
Thanks a lot for looking into this :)

**Comment by Ziriax** [comment]
My bad, it seems the pivot point is correctly exporting, however, it is a bit counter-intuitive how to rotate objects around their pivot after export.

_Before going into the details, could you share your Maya file with me? Because the exported could still be buggy regarding pivot points_ 

If you open my test scene `CubePivotTest1.ma` in Maya, you will notice that the pivot point of the cuboid is moved to a corner.

When exporting this file, you will get the following `glTF`:

```
   "nodes": [
        {
            "children": [
                1
            ],
            "translation": [
                0.5,
                1.0,
                1.5
            ],
            "scale": [
                1.0,
                2.0,
                3.0
            ],
            "name": "pCube1"
        },
        {
            "mesh": 0,
            "translation": [
                -0.5,
                -0.5,
                -0.5
            ],
            "name": "pCubeShape1"
        }
    ],
```
This structure corresponds to the Maya DAG, creating a glTF node for both the Maya transform and shape node.

Maya2glTF will offset all the vertices of the mesh with the pivot, and counter-translate it again. 

Now to rotate the cube around its pivot, you actually have to rotate `pCubeShape1`, not `pCube1`. That might be counter intuitive. 

I guess I should counter-translate the `pCube1` node, leaving the `pCubeShape1` alone. Then you could just rotate `pCube1`, as you would expect.

Note that Maya2glTF only exports the Maya **rotation pivot**, not the scaling pivot (this is all possible by introducing extra nodes, but I didn't find the time yet for this).

**Comment by Ziriax** [comment]
Indeed, at first sight it seems Maya2glTF is buggy when exporting pivot points in a chain of nodes.

For now, I suggest you use Maya's `Modify/Freeze Transformations` to reset all pivot points to zero, then export.

It would still be useful if you could provide me your Maya scene.

**Comment by microcipcip** [comment]
Freeze transformations is the first thing I've tried. [This is my file](https://www.dropbox.com/s/affh2s0ewnvz97m/test2.ma?dl=0), it is a very basic body shape with an arm and hand, I am basically trying to move the whole hand while keeping it attached to the wrist.

**Comment by Ziriax** [comment]
After investigation the way Maya2glTF currently deals with pivot points is flawed. The only thing the exporter currently does is offset the mesh vertices so that a vertex at the pivot point has coordinates (0,0,0). But does not correctly handle a chain of transforms with pivots.

This is not that obvious to do, since the glTF format doesn't have a concept of pivot points.

So to deal with pivot points, I actually have to create **two** nodes for each Maya transform node that has a pivot point.

Luckily I already have this very custom code in place to deal with Maya's joint segment scale compensation, so I will try to "abuse" this code to correctly deal with pivots.

I guess your goal is to manipulate the rotation of each object in code?

Freezing the transforms should work, I will take a look at your scene, and see what can be done, or what goes wrong.

_Many thanks for reporting this issue!_

**Comment by Ziriax** [comment]
I have a fix ready, however it can have a large impact on existing scenes, so I have to test it more before I release it.

To get your scene working, you have to:

1. make a suitable hierarchy
2. bake all pivots
3. export 

[Here you can download the modified Maya scene for your convenience](https://dl.dropboxusercontent.com/s/6aeyfcicu7grhz1/RobotChainTest1.ma)

[Here's a working codepen](https://codepen.io/Ziriax/pen/yqmpWP)

**Comment by microcipcip** [comment]
You are amazing, thanks a lot for your help and for the sample file :)

**Comment by Ziriax** [comment]
This should be fixed by [V0.9.8-alpha](../releases/tag/V0.9.8-alpha) (224bdcad245f1f1fde674e5fc1f24235901fff13)
